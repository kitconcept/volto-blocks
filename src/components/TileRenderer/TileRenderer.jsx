import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { settings, tiles } from '~/config';

/**
 * TileRenderer container class.
 * @class Form
 * @extends Component
 */
class TileRenderer extends Component {
  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    edit: PropTypes.bool.isRequired,
    type: PropTypes.string.isRequired,
    tile: PropTypes.string.isRequired,
    onChangeTile: PropTypes.func,
    data: PropTypes.objectOf(PropTypes.any).isRequired,
  };

  /**
   * Constructor
   * @method constructor
   * @param {Object} props Component properties
   * @constructs Form
   */
  constructor(props) {
    super(props);
    this.state = {};
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    const EditTile = tiles.defaultTilesEditMap[this.props.type];
    const ViewTile = tiles.defaultTilesViewMap[this.props.type];

    if (!this.props.edit) {
      return <ViewTile {...this.props} onChangeTile={() => {}} />;
    }
    if (this.props.edit) {
      return (
        <EditTile
          {...this.props}
          detached
          index={0}
          onSelectTile={() => {}}
          onFocusPreviousTile={() => {}}
          onFocusNextTile={() => {}}
          onAddTile={() => {}}
          onDeleteTile={() => {}}
          onMutateTile={() => {}}
        />
      );
    }
    return '';
  }
}

export default TileRenderer;
