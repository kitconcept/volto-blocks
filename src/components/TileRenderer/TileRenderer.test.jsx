import React from 'react';
import renderer from 'react-test-renderer';
import TileRenderer from './TileRenderer';
import { Provider } from 'react-intl-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();

global.__SERVER__ = true; // eslint-disable-line no-underscore-dangle

test('renders a text tile component using the tile renderer', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
  });
  const component = renderer.create(
    <Provider store={store}>
      <TileRenderer
        data={{ url: 'heroimage.jpg' }}
        tile="thetileid"
        edit
        type="text"
        selected={false}
        onChangeTile={() => {}}
      />
    </Provider>,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
