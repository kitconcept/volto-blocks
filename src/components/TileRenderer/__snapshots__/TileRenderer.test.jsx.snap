// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a text tile component using the tile renderer 1`] = `<div />`;
